src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticService.cs

```diff
@@ -188,25 +197,25 @@ private bool ClearDiagnosticsReportedBySource(IDiagnosticUpdateSource source, Li
             }
         }
 
-        private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
+        private void OnDiagnosticsUpdated(object? sender, ImmutableArray<DiagnosticsUpdatedArgs> e)
         {
-            AssertIfNull(e.Diagnostics);
+            AssertIfNull(e.SelectManyAsArray(e => e.Diagnostics));
```

Comment:
>AssertIfNull [](http://example.com/codeflow?start=12&length=12)

Is this right?

Reply:
➡️ Yes, this one is correct (original behavior has been preserved). This is a helper method that asserts each element of `Diagnostics` is non-null. It assumes (without asserting) that `Diagnostics` itself is non-null.

Reply:
I see now, I was confused around this being an extension method on ImmutableArray